acstools/acsphotcte.py

```diff
@@ -0,0 +1,289 @@
+                        '.amazonaws.com/dev/run-photcte-correction'
+        # Public API ID for the ACS Photometric CTE API
+        self._api_id = ''
+        # Public API key for the ACS PHotometric CTE API
+        self._api_key = ''
+        self._cte_corrections = None
+        self._corrected_fluxes = None
+
+    @property
+    def corrected_fluxes(self):
+        """CTE corrected fluxes"""
+        return self._corrected_fluxes
+
+    @corrected_fluxes.setter
+    def corrected_fluxes(self, value):
+        self._corrected_fluxes = value
+
+    @property
+    def cte_corrections(self):
+        """Computed CTE corrections"""
+        return self._cte_corrections
+
+    @cte_corrections.setter
+    def cte_corrections(self, value):
+        self._cte_corrections = value
+
+    def _query(self, data, api_credentials):
+        """ Submit a post request to the API
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing the request payload
+
+        Returns
+        -------
+        content :
+            Body of the response object
+        """
+        response = requests.post(
+            self._api_url,
+            data=json.dumps(data),
+            headers=api_credentials
+        )
+
+        status_code = response.status_code
+        content = json.loads(response.content)
+        if status_code > 400:
+            content = content['message']
+        return content
+
+    def _check_inputs(self, **inputs):
+        """Check the data types of the user supplied inputs
+
+        JSON is very finicky and the native python json module only works
+        with native python types (int, float, list, str). Hence, before
+        we serialize the inputs as a JSON object we need to compare all the
+        input data types to their python counterparts.
+
+        Parameters
+        ----------
+        inputs : dict
+            Dictionary containing all of the user supplied inputs
+
+        Returns
+        -------
+        inputs : dict
+            The original dictionary with all of the inputs converted to their
+            python counterparts for JSON serialization
+        """
+        for key in inputs.keys():
+            # Check if the input is Iterable and not a list  (e.g. np.array)
+            # We also need to make sure the input is not a str, since str are
+            # Iterable in Python.
+            if isinstance(inputs[key], Iterable) and \
+                    not isinstance(inputs[key], list) and \
+                    not isinstance(inputs[key], str):
+                # Convert the array to a python list
+                inputs[key] = list(inputs[key])
+
+                # Check if the first element of the input array is a float
+                if not isinstance(inputs[key][0], float):
+                    LOG.info('Converting elements..')
+                # Convert the array of elements to floats
+                    inputs[key] = list(map(float, inputs[key]))
```

Comment:
Doesn't this effectively modify `inputs` in-place? Do you still have to return it below?

Reply:
The modified `inputs` is only present in the scope of that method right?

Reply:
`inputs` is a dictionary, so I thought it would be mutable. You can check for yourself.

Reply:
I think I still need to return it.
```console
In [6]: def change_values(**inputs):
   ...:     for key in inputs.keys():
   ...:         print(f'Changing the value {key}')
   ...:         inputs[key]='changed!'
   ...:
   ...: def test():
   ...:     d = {'a':[1,2,3,4],
   ...:         'b':'original'}
   ...:
   ...:     change_values(**d)
   ...:     print(d)
   ...:

In [7]: test()
Changing the value a
Changing the value b
{'a': [1, 2, 3, 4], 'b': 'original'}

In [8]: d = {'a':1234,'b':'1234'}

In [9]: change_values(**d)
Changing the value a
Changing the value b

In [10]: d
Out[10]: {'a': 1234, 'b': '1234'}```

Reply:
Ah, maybe because you passed it in with `**`. Nvm then. Thanks for checking.